src/lib/definitions.js

```diff
@@ -31,6 +22,10 @@ export function validateDefinition (definition, parserType) {
 }
 
 export async function parseDefinition ({ definition, parserType, navSort }) {
+  if (typeof definition === 'string') {
+    definition = YAML.safeLoad(definition)
```

Comment:
No reason why you can't give it pre-parsed structures, for efficiency. The type check is trivial.